jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -298,6 +299,73 @@ public EndPoint getEndPoint()
         return _endPoint;
     }
 
+    /**
+     * <p>Return the Local Name of the connected channel.</p>
+     *
+     * <p>
+     * This is the name after the connector is bound and the connection is accepted.
+     * The value is typically the local address and optionally the local host name
+     * if the local address is unavailable.
+     * </p>
+     * <p>
+     * Value can be overridden by {@link AbstractConnector#setLocalAuthority(HostPort)} for
+     * scenarios where Jetty is being an intermediary and the local name
+     * needs to be changed to satisfy public (pre-intermediary) HTTP behaviors
+     * such as absolute-URI creation (eg: Location response header).
+     * </p>
+     *
+     * @return the local name if overridden, or the local address, or
+     * null in the case of no local address (usually seen in connectors not based on IP networking).
+     */
+    public String getLocalName()
+    {
+        Connector connector = getConnector();
+
+        if (connector instanceof AbstractConnector)
+        {
+            HostPort localAuthority = ((AbstractConnector)connector).getLocalAuthority();
+            if (localAuthority != null)
+                return localAuthority.getHost();
+        }
+
+        InetSocketAddress local = getLocalAddress();
+        if (local != null)
+            return local.getHostString();
+
```

Comment:
We should never fall through to here.  If there `Connector` is a `NetworkConnector` it should always provide a local authority, otherwise return null.

Reply:
See comment above about this feature not being NetworkConnector specific.